app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginFragment.java

```diff
@@ -0,0 +1,154 @@
+package ar.com.wolox.android.training.ui.training.login;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Handler;
+import android.view.View;
+import android.widget.Button;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+
+import ar.com.wolox.android.R;
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment;
+import pl.droidsonroids.gif.GifImageView;
+
+/**
+ * LoginFragment: initial screen with animation and login settings
+ */
+public class LoginFragment extends WolmoFragment<LoginPresenter> implements View.OnClickListener, ILoginView {
+
+    private final long delay = 5000L;
+    private final String urlTyc = "https://www.wolox.com.ar/";
+
+    private View view;
+    private Context ctx;
+
+    private RelativeLayout mContentView;
+    private GifImageView mLogoGif;
+    private TextInputEditText mEmailTxt;
+    private TextInputEditText mPassTxt;
+
+    @Override
+    public int layout() {
+        return R.layout.fragment_login;
+    }
+
+    @Override
+    public void init() {
+
+        ctx = getContext();
+
+        view = getView();
+        if (view != null) {
+            //Views
+            mContentView = view.findViewById(R.id.container_view);
+            mContentView.setVisibility(View.GONE);
+            mLogoGif = view.findViewById(R.id.splash_animation);
+            mLogoGif.setVisibility(View.VISIBLE);
+            mEmailTxt = view.findViewById(R.id.user_text);
+            mPassTxt = view.findViewById(R.id.pass_text);
+
+            // Buttons
+            Button mBtnLogin = view.findViewById(R.id.btn_login);
+            mBtnLogin.setOnClickListener(this);
+            Button mBtnSingup = view.findViewById(R.id.btn_singup);
+            mBtnSingup.setOnClickListener(this);
+            TextView mTvTyc = view.findViewById(R.id.tyc_txt);
+            mTvTyc.setOnClickListener(this);
+        }
+
+        new Handler().postDelayed(this::initMainScreen, delay);
+    }
+
+    private void initMainScreen() {
+        // Hide animation and show main screen
+        mContentView.setVisibility(View.VISIBLE);
+        mLogoGif.setVisibility(View.GONE);
+    }
+
+    @Override
+    public void onClick(View viewOnClick) {
+        // Memory optimization: instead of create one click listener from each button,
+        // we created a global oneClick method for the activity
+        switch (viewOnClick.getId()) {
+            case R.id.btn_login:
+                getPresenter().onLoginButtonClicked(mEmailTxt.getText(), mPassTxt.getText());
+                break;
+            case R.id.btn_singup:
+                toSingUpScreen();
```

Comment:
Remember on MVP the `view` is stupid, it doesn't knows what to do. It should inform the `presenter` the event `onSignupButtonClicked()` and then the `presenter` should say what to do to the view `goToSignUp()`.